common/workunit/workunit.cpp

```diff
@@ -2657,85 +2660,192 @@ cost_type aggregateCost(const IConstWorkUnit * wu, const char *scope, bool exclu
         }
         return totalCost;
     }
+};
+
+GlobalStatisticCollection::GlobalStatisticCollection() : aggregateKindsMapping(aggregateKindStatistics)
+{
+    // Construct statsCollection here as GlobalStatisticCollection::load() is optional
+    StatsScopeId globalScopeId(SSTglobal, (unsigned)0);
+    statsCollection.setown(createStatisticCollection(nullptr, globalScopeId));
 }
 
-//aggregate disk costs from top-level subgraphs (when scope specified) or workflows (scope not specified)
-cost_type aggregateDiskAccessCost(const IConstWorkUnit * wu, const char *scope)
+void GlobalStatisticCollection::load(IConstWorkUnit &workunit, const char * graphName, bool aggregatesOnly)
 {
-    WuScopeFilter filter;
-    if (!isEmptyString(scope))
-        filter.addScope(scope);
-    else
-        filter.addScope("");      // Needed to match scope
-    // when scope is a workflow, sum graph costs (or subgraph cost when no graph cost) to get workflow cost
-    // (Costs from child graphs and activities should have been summed up to graph/subgraph level already)
-    // when isEmptyString(scope), sum workflow costs (or graph cost when no workflow cost) to get global cost
-    // (Costs from all levels below graph should be summed upto at least graph level already)
-    // i.e. need 2 levels of nesting
-    filter.setIncludeNesting(2);
-    // includeNesting(2) needs just source "global". However, WuScopeFilter is incorrectly inferring the source as "global,stats",
-    // causing too many of the stats to be pulled in and inefficiency.  Here, explicitly set source to "global"
-    filter.addSource("global");
-    filter.addOutputStatistic(StCostFileAccess);
-    filter.addRequiredStat(StCostFileAccess);
-    filter.finishedFilter();
-    Owned<IConstWUScopeIterator> it = &wu->getScopeIterator(filter);
-    cost_type totalCost = 0;
-    for (it->first(); it->isValid(); )
+    const char * _wuid = workunit.queryWuid();
```

Comment:
It isn't at the moment.  However, it may be needed later.  For example, when a graph is resumed/restarted, the previous sg stats may be needed.  The costs from previous runs may need to be preserved and aggregated to existing costs.

Reply:
agree, I think aggregateOnly is only ever going to be true for the foreseeable future, and it would be good to remove this option and simplify the code.